codalab/server/bundle_manager.py

```diff
@@ -445,11 +439,19 @@ def get_sort_key(worker):
             else:
                 deps = set(worker['dependencies'])
                 num_available_deps = len(needed_deps & deps)
-            worker_id = worker['worker_id']
 
-            # if the bundle doesn't request GPUs (only request CPUs), prioritize workers that don't have GPUs
-            gpu_priority = bundle_resources.gpus or not has_gpu[worker_id]
-            return (gpu_priority, num_available_deps, worker['cpus'], random.random())
+            # GPU priority is useful when sorting bundles that request for only CPUs. In this case, we want
+            # to rank CPU workers as high as possible and GPU workers as low as possible, so that GPU workers
+            # could be better utilized towards GPU oriented jobs.
+            gpu_priority = worker['gpus'] == 0
+
+            # If the bundle requested for GPUs, prioritize only those workers that have GPUs.
+            # If the bundle didn't request for GPUs (a CPU only job), prioritize all workers based criteria as follows:
+            # 1. prioritize workers that don't have GPUs first
```

Comment:
Be consistent - each of these bullet points should be in the form 'Prioritize workers that ...'.